docs/ccoctl.md

```diff
@@ -161,3 +161,98 @@ $ ccoctl gcp delete --name=<name> --project=<gcp-project-id>
+4. Use the step 3 ram user’s accesskey id/secret to configure the Alibaba Cloud SDK client's creadential provider chain with [Envionment Creadentials](https://github.com/aliyun/alibaba-cloud-sdk-go/blob/master/docs/2-Client-EN.md#1-environment-credentials) mode or through [Credentials File](https://github.com/aliyun/alibaba-cloud-sdk-go/blob/master/docs/2-Client-EN.md#2-credentials-file) mode
+
+### Procedure
+
+1. Extract the list of CredentialsRequest custom resources (CRs) from the OpenShift Container Platform release image:
+
+   ```bash
+   $ oc adm release extract --credentials-requests --cloud=alibabacloud --to=<path_to_directory_with_list_of_credentials_requests>/credrequests quay.io/<path_to>/ocp-release:<version>
+   
+   ```
+
+**step 2&3 are only need when preparing for upgrading clusters with manually maintained credentials. When doing a fresh installation please skip step 2&3**
+
+2. For each CredentialsRequest CR in the release image, ensure that a namespace that matches the text in the spec.secretRef.namespace field exists in the cluster. This field is where the generated secrets that hold the credentials configuration are stored.
+
+   Sample Alibaba Cloud CredentialsRequest object
+
+   ```yaml
+   apiVersion: cloudcredential.openshift.io/v1
+   kind: CredentialsRequest
+   metadata:
+     name: cloud-credential-operator-ram-ro
+     namespace: openshift-cloud-credential-operator
+   spec:
+     providerSpec:
+       apiVersion: cloudcredential.openshift.io/v1
+       kind: AlibabaCloudProviderSpec
+       statementEntries:
+       - action:
+         - ecs:CopySnapshot
+         - ecs:DeleteDisk
+         - ecs:DescribeInstanceAttribute
+         - ecs:DescribeInstances
+         effect: Allow
+         resource: '*'
+     secretRef:
+       namespace: cloud-credential-operator-ram-ro-creds
+       name: openshift-cloud-credential-operator
+   ```
+
+ 3. For any `CredentialsRequest` CR for which the cluster does not already have a namespace with the name specified in `spec.secretRef.namespace`, create the namespace:
+
+   ```
+   $ oc create namespace <component_namespace>
+   ```
+
+4. Use the `ccoctl` tool to process all `CredentialsRequest` objects in the `credrequests` directory:
+
+   ```bash
+   $ ccoctl alibabacloud attach-ram-policy --name <name> --region=<region> --credentials-requests-dir=<path_to_directory_with_list_of_credentials_requests>/credrequests --output-dir=xxxxxx
+   ```
+
+    where:
+
+   - `name` is the name used to tag any cloud resources that are created for tracking. 
+   - `region` is the Alibaba Cloud region in which cloud resources will be created.
+   - `credentials-requests-dir` is the directory containing files of component CredentialsRequests.
+   - `output-dir`/manifests is the directory containing files of component credentials secret.
```

Comment:
Docs are inconsistent. I think command `attach-ram-policy` has changed to `attach-ram-policy-to-existing-user` now.